prime-router/src/main/kotlin/JurisdictionalFilters.kt

```diff
@@ -157,7 +161,112 @@ class OrEquals : JurisdictionalFilter {
+}
+
+
+/**
+ * Implements a quality check match.  If a row has valid data for all the columns, the row is selected.
+ * If any column name does not exist, nothing passes thru the filter.
+ * hasValidDataFor(columnName1, columnName2, columnName3, ...)
+ * If no columns are passed, all rows are selected.  So, any number of args is acceptable.
+ */
+class HasValidDataFor : JurisdictionalFilter {
+    override val name = "hasValidDataFor"
+
+    override fun getSelection(args: List<String>, table: Table, receiver: Receiver): Selection {
+        var selection = Selection.withRange(0, table.rowCount())
+
+        val columnNames = table.columnNames()
+        args.forEach {
+            if (columnNames.contains(it)) {
+                selection = selection.andNot(table.stringColumn(it).isEmptyString)
+            } else {
+                logger.warn("Report for ${receiver.fullName} does not contain column $it." +
+                    "  All data in this report will fail the quality check")
+                return Selection.withRange(0, 0)
+            }
+        }
+        return selection
+    }
+}
+
+/**
+ * Implements a quality check match.  If a row has valid data for all the columnsm, the row is selected.
+ * If any column name does not exist, nothing passes thru the filter.
+ * hasValidDataFor(columnName1, columnName2, columnName3, ...)
+ * If no columns are passed, all rows are selected.  So, any number of args is acceptable.
+ */
+class HasAtLeastOneOf : JurisdictionalFilter {
+    override val name = "hasAtLeastOneOf"
+
+    override fun getSelection(args: List<String>, table: Table, receiver: Receiver): Selection {
+        if (args.isEmpty()) error("Expecting at least one arg for filter $name.  Got none.")
+        var selection = Selection.withRange(0, 0)
+        val columnNames = table.columnNames()
+        var atLeastOneColumnFound = false
+        args.forEach {
+            if (columnNames.contains(it)) {
+                selection = selection.or(table.stringColumn(it).isNotMissing)
+                atLeastOneColumnFound = true
+            }
+        }
+        if (!atLeastOneColumnFound) {
+            logger.warn("Report for ${receiver.fullName} does not contain any of these columns:" +
+                " ${args.joinToString(",")}" +
+                ".  All data in this report will fail the quality check")
+        }
+        return selection
+    }
+}
+
+
+
+
 object JurisdictionalFilters {
+    // covid-19 default quality check consists of these filters
+    // todo move this to a GLOBAL Setting in the settings table
+    val defaultCovid19QualityCheck = listOf(
+        // valid human and valid test
+        "hasValidDataFor(" +
+            "message_id," +
+            "equipment_model_name," +
+            "specimen_type," +
+            "test_result," +
+            "patient_last_name," +
+            "patient_first_name," +
+            "patient_dob" +
+        ")",
+        // has valid location (for contact tracing)
```

Comment:
This might read cleaner if you do a multiline string instead of concatenation.

Reply:
Good point, although I'm not sure our regex would swallow that, and didn't want to spend time testing it.